src/pybind/mgr/cephadm/agent.py

```diff
@@ -57,19 +58,19 @@ def configure_routes(self) -> None:
         cherrypy.tree.mount(self.node_proxy_endpoint, '/node-proxy', config=conf)
 
     def configure_tls(self, server: Server) -> None:
-        old_cert = self.mgr.cert_key_store.get_cert('agent_endpoint_root_cert')
-        old_key = self.mgr.cert_key_store.get_key('agent_endpoint_key')
+        addr = self.mgr.get_mgr_ip()
+        host = self.mgr.get_hostname()
+        cert, key = self.mgr.cert_mgr.generate_cert(host, addr)
```

Comment:
should this cert, which isn't quite the agent endpoint root cert any more, but is still a cert we're making use of, be kept around? I think with the agent enabled if we don't store the cert we have to reconfigure every agent's individual certificate on mgr failover, so there could be cause for storing it (and l don't know what the downside would be)

Reply:
mm I would say the behavior is as the previous code the only difference is that the agent doesn't have its own root CA, instead cephadm root CA is used and it's stored in the key-store. We are still generating certs on the file as we did in the previous code. Unless there's a real need to store the agent certificate around I'd suggest keeping the same behavior. 

> I think with the agent enabled if we don't store the cert we have to reconfigure every agent's individual certificate on mgr failover

Since the behavior has not changed if this is an issue it should be already happening with the current code. I'd recommend testing the agent to see if this is broken (or not). If it's the case open a BUG tracker and fix it on a separate PR just to make sure the BUG is documented. WDYT?

Reply:
Alright,  teuthology includes tests that use the agent, so we'll see when I run it again with your newest changes